runtime/doc/lua.txt

```diff
@@ -2464,6 +2464,73 @@ parents({start})                                            *vim.fs.parents()*
         (function) Iterator
 
 
+==============================================================================
+Lua module: func                                                    *lua-func*
+
+on_fun({container}, {key}, {fn})                           *vim.func.on_fun()*
+    Sets function `container[key]` to a new (wrapper) function that calls
+    `fn()` before optionally calling the original ("base") function.
+
+    The result of `fn()` decides how the base function is invoked. Given `fn()` : >lua
```

Comment:
This "control the behavior via return values", probably isn't needed.

Instead `fn()` could simply take the overridden function as the first arg, and the parameters are in a `table.pack()` table named `args`.  Then everything is much more natural.

```
function(overridden, args)
```